pom.xml

```diff
@@ -215,25 +215,74 @@
       <artifactId>script-security</artifactId>
       <version>1.54</version>
     </dependency>
-    <!-- not sure if this makes any difference but attempting to fix enforcer errors-->
-    <!-- TODO rather use dependencyManagement if necessary -->
-    <dependency>
-      <groupId>joda-time</groupId>
-      <artifactId>joda-time</artifactId>
-      <version>2.10.1</version>
-    </dependency>
-    <dependency>
-      <groupId>io.fabric8</groupId>
-      <artifactId>kubernetes-client</artifactId>
-      <version>4.1.3</version>
-    </dependency>
-    <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-annotations</artifactId>
-      <version>2.9.0</version>
-    </dependency>
   </dependencies>
 
+  <!--
+    This is to just remove the UpperBounds rule
+    This rule is covered by the tests that actually start jenkins and look for plugin start failures.
```

Comment:
`InjectedTest`—did you check that this is in fact run during PR builds of this component?

Reply:
@jglick not `InjectedTest` but the test that runs the resultant war with a pipeline.
the test that checks the logs actually needs to be written and added to [`jx/scripts/test.sh`](https://github.com/jenkins-x/jenkins-x-serverless-filerunner/blob/0f03af96d6ad4217c7fdb8ff0baf19ccd1cfb437/jx/scripts/test.sh) and hence why this PR is on hold.
and yes this is run on a PR

Reply:
> not `InjectedTest` but the test that runs the resultant war with a pipeline

That will _not_ verify that plugin dependencies are correct: if something is broken, `PluginManager` will decide that the depending plugin should be disabled and log a warning…but then `ExtensionList` etc. will load its features anyway because they can be discovered on the `java.class.path`. That is why you need to run `InjectedTest`, which specifically fails if it finds any disabled plugins.

Reply:
Hi @jglick, have you got any details on how `InjectedTest` runs, or how to enable it?

Reply:
It would be generated during a test phase if you have `<packaging>hpi</packaging>`, which this is not. Lacking that, `requireUpperBoundDeps` is another (much faster) line of defense, though it can get confused by some special cases—for example that all plugins declare a Jenkins core version no newer than the one declared here (since `provided` deps are excluded from the dependency chain).